packages/orchestrator/lib/clients/client.ts

```diff
@@ -0,0 +1,264 @@
+            }
+        };
+        return this.execute(schedulingProps);
+    }
+
+    public async executeWebhook(props: ExecuteWebhookProps): Promise<ExecuteReturn> {
+        const { args, ...rest } = props;
+        const schedulingProps = {
+            ...rest,
+            args: {
+                ...args,
+                type: 'webhook' as const
+            }
+        };
+        return this.execute(schedulingProps);
+    }
+
+    public async search({
+        ids,
+        groupKey,
+        limit
+    }: {
+        ids?: string[];
+        groupKey?: string;
+        limit?: number;
+    }): Promise<Result<(TaskWebhook | TaskAction)[], ClientError>> {
+        const body = {
+            ...(ids ? { ids } : {}),
+            ...(groupKey ? { groupKey } : {}),
+            ...(limit ? { limit } : {})
+        };
+        const res = await this.routeFetch(postSearchRoute)({ body });
+        if ('error' in res) {
+            return Err({
+                name: res.error.code,
+                message: res.error.message || `Error listing tasks`,
+                payload: body
+            });
+        } else {
+            const tasks = res.flatMap((task) => {
+                const validated = validateTask(task);
+                if (validated.isErr()) {
+                    return [];
+                }
+                return [validated.value];
+            });
+            return Ok(tasks);
+        }
+    }
+
+    public async dequeue({
+        groupKey,
+        limit,
+        waitForCompletion = true
+    }: {
+        groupKey: string;
+        limit: number;
+        waitForCompletion?: boolean;
+    }): Promise<Result<(TaskWebhook | TaskAction)[], ClientError>> {
+        const res = await this.routeFetch(getDequeueRoute)({
+            query: {
+                groupKey,
+                limit,
+                waitForCompletion
+            }
+        });
+        if ('error' in res) {
+            return Err({
+                name: res.error.code,
+                message: res.error.message || `Error dequeueing tasks`,
+                payload: { groupKey, limit }
+            });
+        } else {
+            const dequeuedTasks = res.flatMap((task) => {
+                const validated = validateTask(task);
+                if (validated.isErr()) {
+                    return [];
+                }
+                return [validated.value];
+            });
+            return Ok(dequeuedTasks);
+        }
+    }
+
+    public async heartbeat({ taskId }: { taskId: string }): Promise<Result<void, ClientError>> {
+        const res = await this.routeFetch(postHeartbeatRoute)({
+            params: { taskId }
+        });
+        if ('error' in res) {
+            return Err({
+                name: res.error.code,
+                message: res.error.message || `Error heartbeating task '${taskId}'`,
+                payload: { taskId }
+            });
+        } else {
+            return Ok(undefined);
```

Comment:
should this be `null`?

Reply:
`null` is not assignable to type `void`. Only `undefined` is. 🤷